internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -126,7 +126,21 @@ func (ws *workflows) StartWorkflow(ctx context.Context, session sdktypes.Session
 
 func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkflowParams) error {
 	wi := workflow.GetInfo(wctx)
-	sid := params.Data.Session.ID()
+	session := params.Data.Session
+	sid := session.ID()
+
+	// get session time (it was created not long after event dispatch) and then update it with actual event time if found
```

Comment:
Please add punctuation and/or improve the wording, I'm not sure what this comment is saying.

Also, you have to handle and report errors in the block below, you can't assume that the data is correct or that the code will work as expected.

Reply:
1. removed the comment - it was remaining of the old flow
2. there is no handling error in the block below. We are trying to extract event time and if we fail, we should continue with the session flow. The only thing that should happen is that metric won't be reported (per efi's request. I would rather to report this metric with session time instead if there is no event time - this what comment was saying)